aws-opensearchserverless-lifecyclepolicy/src/main/java/software/amazon/opensearchserverless/lifecyclepolicy/UpdateHandler.java

```diff
@@ -0,0 +1,140 @@
+                // NOTE: If your service API throws 'ResourceNotFoundException'
+                // for update requests this method is not necessary
+                proxy.initiate("AWS-OpenSearchServerless-LifecyclePolicy::Update::PreUpdateCheck", proxyClient,
+                        progress.getResourceModel(), progress.getCallbackContext())
+
+                    // STEP 1.1 [initialize a proxy context]
+                    .translateToServiceRequest(Translator::translateToReadRequest)
+
+                    // STEP 1.2 [make an api call]
+                    .makeServiceCall((awsRequest, client) -> {
+                        // add custom read resource logic
+                        // If describe request does not return ResourceNotFoundException,
+                        // you must throw ResourceNotFoundException based on
+                        // awsResponse values
+                        BatchGetLifecyclePolicyResponse awsResponse = batchGetLifecyclePolicy(awsRequest, client, logger);
+                        callbackContext.setCurrentLifecyclePolicyDetail(awsResponse.lifecyclePolicyDetails().get(0));
+                        return awsResponse;
+                    })
+                    .progress()
+            )
+            .then(progress ->
+                // STEP 2.0 [initialize a proxy context]
+                // Implement client invocation of the update request through the proxyClient,
+                // which is already initialised with
+                // caller credentials, correct region and retry settings
+                proxy.initiate("AWS-OpenSearchServerless-LifecyclePolicy::Update",
+                        proxyClient, request.getDesiredResourceState(), callbackContext)
+                    .translateToServiceRequest(resourceModel -> Translator.translateToUpdateRequest(resourceModel,
+                        callbackContext.getCurrentLifecyclePolicyDetail()))
+                    .makeServiceCall((awsRequest, client) -> updateLifecyclePolicy(awsRequest, client, logger))
+                    .done(awsResponse -> ProgressEvent.defaultSuccessHandler(
+                        Translator.translateFromUpdateResponse(awsResponse))));
+    }
+
+    private UpdateLifecyclePolicyResponse updateLifecyclePolicy(final UpdateLifecyclePolicyRequest updateLifecyclePolicyRequest,
+                                                                final ProxyClient<OpenSearchServerlessClient> proxyClient,
+                                                                final Logger logger) {
+        UpdateLifecyclePolicyResponse updateLifecyclePolicyResponse;
+        try {
+            updateLifecyclePolicyResponse = proxyClient.injectCredentialsAndInvokeV2(updateLifecyclePolicyRequest, proxyClient.client()::updateLifecyclePolicy);
+        } catch (ResourceNotFoundException e) {
+            throw new CfnNotFoundException(e);
+        } catch (ValidationException e) {
```

Comment:
Does not update also throw Conflict exception?

Reply:
It will throw Conflict Exception